numpy/lib/tests/test_io.py

```diff
@@ -1823,6 +1828,123 @@ def test_auto_dtype_largeint(self):
         assert_equal(test['f1'], 17179869184)
         assert_equal(test['f2'], 1024)
 
+
+class TestPathUsage(TestCase):
+    @contextlib.contextmanager
```

Comment:
except for our tempdir context manager being broken this looks fine to me
fixing tempdir can be in a separate PR

though one could also go the route of creating a tempdir in the TestCase.setUp and delete it in tearDown, it seems each test only wants one file anyway